Src/Plugins/Security/SEC101_050.IdentifiableNpmAuthorTokenValidator.cs

```diff
@@ -0,0 +1,58 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+using System.Net.Http;
+
+using Microsoft.CodeAnalysis.Sarif.PatternMatcher.Sdk;
+using Microsoft.RE2.Managed;
+using Microsoft.Security.Utilities;
+
+namespace Microsoft.CodeAnalysis.Sarif.PatternMatcher.Plugins.Security
+{
+    public class IdentifiableNpmAuthorTokenValidator : DynamicValidatorBase
+    {
+        private readonly CustomAlphabetEncoder encoder = new CustomAlphabetEncoder();
```

Comment:
>new CustomAlphabetEncoder(); [](http://example.com/codeflow?start=57&length=28)

can you move this to the constructor of this class?
we try to move initializations to the constructor.

Reply:
Good note, please provide the rationale for the standard. 'The reason is' that it's harder to understand initialization behavior when assignments are inlined like this: instance variables can be declared anywhere in the class. You really notice the problem with this pattern when debugging a ctor with initializations of this kind that are scattered around: the debugger hopes from place to place. With an explicit constructor, everything is sequential.